packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -54,12 +53,14 @@ const Search = props => {
     }
   };
 
-  const loadAlgoliaJS = () => {
-    if (!loadedJs) {
-      loadJS().then(a => {
-        loadedJs = true;
-        window.docsearch = a.default;
-        initAlgolia();
+  const loadAlgolia = () => {
+    if (!loaded) {
+      loadCss().then(() => {
```

Comment:
Why waterfall downloading here? They are not dependent so it can be parallel? 

```js
Promise.all([loadCss, loadJS]).then(([a, _]) => ...);
```

Otherwise, is it possible to change to `await`? In the very worst case, the nested `.then()` is not necessary. I think you can change to:

```js
loadCss().then(loadJS).then(a => { ... });
```